src/webview/memory-webview-view.tsx

```diff
@@ -212,8 +217,16 @@ class App extends React.Component<{}, MemoryAppState> {
         this.fetchMemory();
     }, 100);
 
-    protected sessionContextChanged(sessionContext: SessionContext): void {
-        this.setState({ sessionContext });
+    protected sessionContextChanged(allContext: [SessionContext, Context?, Context[]?]): void {
+        const [sessionContext, currentContext, contexts] = allContext;
+        const updatedState: Partial<MemoryAppState> = {};
+        if (typeof (currentContext) !== 'undefined') {
+            updatedState.context = currentContext;
+        }
+        if (contexts?.length) {
+            updatedState.contexts = contexts;
+        }
+        this.setState(prevState => ({ ...prevState, sessionContext, ...updatedState }));
```

Comment:
This raises the question of the direction of flow for the new `Context` values.
 - On the one hand, the user is allowed to make a selection among the available contexts, so there's some FE -> BE flow.
 - On the other hand, only the adapter knows what the available contexts are, and that set may change (new threads, changes to the stack), so there is some BE -> FE flow.

We may want to be more explicit about how we handle changes to the available contexts and what happens if the context that we believe we're in goes away (gets popped off the stack, e.g.).

Reply:
Yes, good point. These should be split apart.